### PR TITLE
Update GA4 naming conventions

### PR DESCRIPTION
**Pull request by JamesCGDS**
## What
The changes in this PR update the naming conventions in our GA4 analytics implementation (e.g. addressing capitalisation in variable names and replacing references to Google Tag Manager with Google Analytics where appropriate). The key changes are:

- Renaming `Ga4EnhancedEcommerceTracker` to `Ga4EcommerceTracker`
- Renaming `GOVUK.analyticsGA4` to `GOVUK.analyticsGa4`
- Replacing instances of `gtm` with `ga4` where appropriate

Not all instances of `gtm` have been replaced with `ga4`. This is because these changes only relate to the GA4 upgrade; this PR does not remove instances of `gtm` from older analytics implementations.

These changes are breaking as `static` and `finder-frontend` are referencing old values (e.g. `GOVUK.analyticsGA4` rather than `GOVUK.analyticsGa4`). [PR #2888](https://github.com/alphagov/finder-frontend/pull/2888) in `finder-frontend` and [PR #2892](https://github.com/alphagov/static/pull/2892) in `static` have been raised to update the values accordingly in these applications.

## Why
To improve the naming conventions in our GA4 analytics implementation.

## Visual Changes
N/A

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@
   useful summary for people upgrading their application, not a replication
   of the commit log.
 
+## Unreleased
+
+* Fix bug when getting target element from URL hash in accordion component ([PR #2985](https://github.com/alphagov/govuk_publishing_components/pull/2985))
```
Comment (andysellick) [inline CHANGELOG.md:12]: Was this missed in a previous PR?

**Review by andysellick — APPROVED** [review]
Ahhh that's better. Good stuff. Only one thing - the last commit is a bit odd, what happened there?

File: CHANGELOG.md (hunk 1 above)
Comment (JamesCGDS) [reply CHANGELOG.md:12]: Yeah, when I added it I was still getting a merge conflict for some reason which resulted in the strange commit at the end.

**Comment by JamesCGDS** [comment]
Thanks @andysellick. Yeah, basically there was a merge conflict with my CHANGELOG so I resolved it in GitHub, which resulted in the strange commit. I think I can squash it into the previous commit though to make things a bit tidier.

File: CHANGELOG.md (hunk 1 above)
Comment (JamesCGDS) [reply CHANGELOG.md:12]: And the order of the commits got a bit jumbled, I'll have a look at sorting it out before I merge.

**Pull request merged**